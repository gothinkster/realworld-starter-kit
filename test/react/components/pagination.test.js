/* eslint-disable no-undef, function-paren-newline */
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Pagination from '../../../src/react/components/pagination';

configure({ adapter: new Adapter() });


test('Paginator snapshot', () => {
  const props = {
    count: 101,
    pageLength: 10,
    page: 10,
  };
  const component = renderer.create(
    <MemoryRouter initialEntries={['/', '/page/10', '/next']} initialIndex={1}>
      <Route path="/page/:page">
        <Pagination {...props} />
      </Route>
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Paginator for root route 1st page', () => {
  const props = {
    count: 101,
    pageLength: 10,
    page: 1,
  };
  const component = mount(
    <MemoryRouter initialEntries={['/before', '/', '/next']} initialIndex={1}>
      <Route path="/">
        <Pagination {...props} />
      </Route>
    </MemoryRouter>,
  );
  expect(component.find('li.active').find('a').prop('href')).toEqual('/');
  expect(component.find('li').first().find('a').prop('href')).toEqual('/');
});


test('Paginator for root route 2nd page', () => {
  const props = {
    count: 101,
    pageLength: 10,
    page: 2,
  };
  const component = mount(
    <MemoryRouter initialEntries={['/', '/page/2', '/next']} initialIndex={1}>
      <Route path="/page/:page">
        <Pagination {...props} />
      </Route>
    </MemoryRouter>,
  );
  expect(component.find('li.active').find('a').prop('href')).toEqual('/page/2');
  expect(component.find('li').first().find('a').prop('href')).toEqual('/');
});


test('Paginator for some route 1st page', () => {
  const props = {
    count: 101,
    pageLength: 10,
    page: 1,
  };
  const component = mount(
    <MemoryRouter initialEntries={['/', '/some', '/next']} initialIndex={1} context={{}}>
      <Route path="/some">
        <Pagination {...props} />
      </Route>
    </MemoryRouter>,
  );
  expect(component.find('li.active').find('a').prop('href')).toEqual('/some');
  expect(component.find('li').first().find('a').prop('href')).toEqual('/some');
});


test('Paginator for /some route 2nd page', () => {
  const props = {
    count: 101,
    pageLength: 10,
    page: 2,
  };
  const component = mount(
    <MemoryRouter initialEntries={['/', '/some/page/2', '/next']} initialIndex={1}>
      <Route path="/some/page/:page">
        <Pagination {...props} />
      </Route>
    </MemoryRouter>,
  );
  expect(component.find('li.active').find('a').prop('href')).toEqual('/some/page/2');
  expect(component.find('li').first().find('a').prop('href')).toEqual('/some');
});
