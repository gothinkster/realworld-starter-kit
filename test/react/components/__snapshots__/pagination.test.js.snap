// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator snapshot 1`] = `
<nav>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/2"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/3"
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/4"
        onClick={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/5"
        onClick={[Function]}
      >
        5
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/6"
        onClick={[Function]}
      >
        6
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/7"
        onClick={[Function]}
      >
        7
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/8"
        onClick={[Function]}
      >
        8
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/9"
        onClick={[Function]}
      >
        9
      </a>
    </li>
    <li
      className="page-item active"
    >
      <a
        className="page-link"
        href="/page/10"
        onClick={[Function]}
      >
        10
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/page/11"
        onClick={[Function]}
      >
        11
      </a>
    </li>
  </ul>
</nav>
`;
